tests/unit/backend/corpora/common/entities/test_dataset_asset.py

```diff
@@ -0,0 +1,84 @@
+import mock
+
+from botocore.exceptions import ClientError
+
+from backend.corpora.common.corpora_orm import (
+    DatasetArtifactType,
+    DatasetArtifactFileType,
+)
+from backend.corpora.common.entities.dataset import Dataset
+from backend.corpora.common.entities.dataset_asset import DatasetAsset
+from tests.unit.backend.corpora import CorporaTestCaseUsingMockAWS
+from tests.unit.backend.utils import BogusDatasetParams
+
+
+class TestDatasetAsset(CorporaTestCaseUsingMockAWS):
+    def setUp(self):
+        super().setUp()
+        self.uuid = "test_dataset_artifact_id"
+        self.bucket_name = self.CORPORA_TEST_CONFIG["bucket_name"]
+
+    def test__get__ok(self):
+        asset = DatasetAsset.get(self.uuid)
+        self.assertEqual(self.uuid, asset.id)
+        self.assertEqual(self.CORPORA_TEST_CONFIG["bucket_name"], asset.bucket_name)
+        self.assertEqual("test_s3_uri.h5ad", asset.file_prefix)
+
+    def test__generate_file_url__OK(self):
+        file_name = "test_generate_file_url.h5ad"
+
+        # Create the Dataset Asset
+        asset = self.create_dataset_asset(file_name)
+
+        url = asset.generate_file_url()
+        self.assertIn(file_name, url)
+        self.assertIn("Expires=", url)
+
+    @mock.patch("backend.corpora.common.entities.dataset_asset.DatasetAsset.s3.generate_presigned_url")
+    def test__generate_file_url__ERROR(self, mock_generate_presigned_url):
+        def side_effect(*args, **kwargs):
+            raise ClientError({}, "mock ClientError")
+
+        mock_generate_presigned_url.side_effect = side_effect
+
+        # Create the Dataset Asset
```

Comment:
I think you want to verify the logs as well because that tell you whether you hit the expected error besides just returning None. So I'd do:

```
asset = self.create_dataset_asset("test__generate_file_url__ERROR.h5ad")

with self.assertLogs(level="ERROR") as logger:
    url = asset.generate_file_url()
self.assertIn("Failed to generate presigned URL", logger.output[0])
self.assertIsNone(url)
```